File: src/org/privatechats/securesms/database/loaders/ConversationLoader.java
package org.privatechats.securesms.database.loaders;

import android.content.Context;
import android.database.Cursor;

import org.privatechats.securesms.database.DatabaseFactory;
import org.privatechats.securesms.util.AbstractCursorLoader;

public class ConversationLoader extends AbstractCursorLoader {
  private final long threadId;
  private       long limit;

  public ConversationLoader(Context context, long threadId, long limit) {
    super(context);
    this.threadId = threadId;
    this.limit  = limit;
  }

  public boolean hasLimit() {
    return limit > 0;
  }

  @Override
  public Cursor getCursor() {
    return DatabaseFactory.getMmsSmsDatabase(context).getConversation(threadId, limit);
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationLoader.kt
package org.thoughtcrime.securesms.conversation.v2

import android.content.Context
import android.database.Cursor
import org.thoughtcrime.securesms.dependencies.DatabaseComponent
import org.thoughtcrime.securesms.util.AbstractCursorLoader

class ConversationLoader(
    private val threadID: Long,
    private val reverse: Boolean,
    context: Context
) : AbstractCursorLoader(context) {

    override fun getCursor(): Cursor {
        return DatabaseComponent.get(context).mmsSmsDatabase().getConversation(threadID, reverse)
    }
}